tests/ExpressionParserTest.php

```diff
@@ -411,6 +422,111 @@ public function testUnknownTestWithoutSuggestions()
+        $env->addFilter(new TwigFilter('foo', 'foo', ['node_class' => NotReadyFilterExpressionWithNoConstructor::class]));
+        $parser = new Parser($env);
+
+        $parser->parse($env->tokenize(new Source('{{ 1|foo }}', 'index')));
+        $this->doesNotPerformAssertions();
+    }
+
+    public function testNotReadyTestWithNoConstructor()
+    {
+        $env = new Environment(new ArrayLoader(), ['cache' => false, 'autoescape' => false]);
+        $env->addTest(new TwigTest('foo', 'foo', ['node_class' => NotReadyTestExpressionWithNoConstructor::class]));
+        $parser = new Parser($env);
+
+        $parser->parse($env->tokenize(new Source('{{ 1 is foo }}', 'index')));
+        $this->doesNotPerformAssertions();
+    }
+
+    /**
+     * @group legacy
+     */
+    public function testNotReadyFunction()
+    {
+        $this->expectDeprecation('Since twig/twig 3.12: Twig node "Twig\Tests\NotReadyFunctionExpression" is not marked as ready for passing a "TwigFunction" instead instead of its name; please make it ready and then flag its constructor with the #[FirstClassTwigCallableReady] attribute.');
+        $this->expectDeprecation('Since twig/twig 3.12: Not passing an instance of "TwigFunction" when creating a "foo" function of type "Twig\Tests\NotReadyFunctionExpression" is deprecated.');
+
+        $env = new Environment(new ArrayLoader(), ['cache' => false, 'autoescape' => false]);
+        $env->addFunction(new TwigFunction('foo', 'foo', ['node_class' => NotReadyFunctionExpression::class]));
+        $parser = new Parser($env);
+
+        $parser->parse($env->tokenize(new Source('{{ foo() }}', 'index')));
+    }
+
+    /**
+     * @group legacy
+     */
+    public function testNotReadyFilter()
+    {
+        $this->expectDeprecation('Since twig/twig 3.12: Twig node "Twig\Tests\NotReadyFilterExpression" is not marked as ready for passing a "TwigFilter" instead instead of its name; please make it ready and then flag its constructor with the #[FirstClassTwigCallableReady] attribute.');
+        $this->expectDeprecation('Since twig/twig 3.12: Not passing an instance of "TwigFilter" when creating a "foo" filter of type "Twig\Tests\NotReadyFilterExpression" is deprecated.');
+
+        $env = new Environment(new ArrayLoader(), ['cache' => false, 'autoescape' => false]);
+        $env->addFilter(new TwigFilter('foo', 'foo', ['node_class' => NotReadyFilterExpression::class]));
+        $parser = new Parser($env);
+
+        $parser->parse($env->tokenize(new Source('{{ 1|foo }}', 'index')));
+    }
+
+    /**
+     * @group legacy
+     */
+    public function testNotReadyTest()
+    {
+        $this->expectDeprecation('Since twig/twig 3.12: Twig node "Twig\Tests\NotReadyTestExpression" is not marked as ready for passing a "TwigTest" instead instead of its name; please make it ready and then flag its constructor with the #[FirstClassTwigCallableReady] attribute.');
```

Comment:
```suggestion
        $this->expectDeprecation('Since twig/twig 3.12: Twig node "Twig\Tests\NotReadyTestExpression" is not marked as ready for passing a "TwigTest" instead of its name; please make it ready and then flag its constructor with the #[FirstClassTwigCallableReady] attribute.');
```